File: ce/snippets/csharp/CRMV8/query/cs/linqexamples3.cs
// <snippetlinqexamples3>


 using (ServiceContext svcContext = new ServiceContext(_serviceProxy))
 {
  var query_where3 = from c in svcContext.ContactSet
                     join a in svcContext.AccountSet
                     on c.ContactId equals a.PrimaryContactId.Id
                     where a.Name.Contains("Contoso")
                     where c.LastName.Contains("Smith")
                     select new
                     {
                      account_name = a.Name,
                      contact_name = c.LastName
                     };

  foreach (var c in query_where3)
  {
   System.Console.WriteLine("acct: " +
    c.account_name +
    "\t\t\t" +
    "contact: " +
    c.contact_name);
  }
 }

// </snippetlinqexamples3>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples3.vb
' <snippetlinqexamples3>


 Using svcContext As New ServiceContext(_serviceProxy)
  Dim query_where3 = From c In svcContext.ContactSet _
                     Join a In svcContext.AccountSet _
                     On c.ContactId Equals a.account_primary_contact.Id _
                     Where a.Name.Contains("Contoso") _
                     Where c.LastName.Contains("Smith") _
                     Select New With {Key .account_name = a.Name,
                                      Key .contact_name = c.LastName}

  For Each c In query_where3
   Console.WriteLine("acct: " &amp; c.account_name &amp; vbTab &amp; vbTab _
                     &amp; vbTab &amp; "contact: " &amp; c.contact_name)
  Next c
 End Using

' </snippetlinqexamples3>